### PR TITLE
Add support for template-version.yml and remove catalog field

### PR DESCRIPTION
**Pull request by amy**
Add support for template-version.yml, removes catalog field for template-version.yml, and only includes the old catalog fields. https://github.com/rancher/rancher/issues/8319

### CODE REVIEW

**Comment by amy** [comment]
cc/ @joshwget @cjellick

File: manager/traverse.go [hunk 1]
```diff
@@ -193,15 +193,30 @@ func traverseGitFiles(repoPath string) ([]model.Template, []error, error) {
...
 			}
 			var rancherCompose string
+			var templateVersion string
 			for _, file := range version.Files {
+
 				if file.Name == "rancher-compose.yml" {
 					rancherCompose = file.Contents
 				}
+
+				if strings.HasPrefix(file.Name, "template-") && strings.HasSuffix(file.Name, ".yml") {
```
Comment (joshwget) [inline manager/traverse.go]: Is there a reason to support `template-*.yml` instead of just `template-version.yml`?

File: parse/compose.go [hunk 2]
```diff
@@ -7,6 +7,22 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
+func CatalogInfoFromTemplateVersion(contents []byte) (model.Version, error) {
+
+	var data map[string]interface{}
```
Comment (joshwget) [inline parse/compose.go]: Assuming the map in between here is for supporting camel case and underscore fields later on.

**Comment by joshwget** [comment]
Can you add an integration test for this?

File: manager/traverse.go (hunk 1 above)
Comment (amy) [inline manager/traverse.go]: Oh. I think I misinterpreted the `version` part and thought that needed to accept like a version. Lol. I'll change that now.

File: parse/compose.go (hunk 2 above)
Comment (amy) [inline parse/compose.go]: Didn't notice this was still here. Its leftover thoughts that I should separate into a different PR. Removing it.

File: manager/traverse.go (hunk 1 above)
Comment (amy) [inline manager/traverse.go]: Ah. Misinterpreted the `version` specification as accepting any version.

File: parse/compose.go (hunk 2 above)
Comment (amy) [inline parse/compose.go]: Oops. I'll remove that since its not useful for this PR.

**Comment by joshwget** [comment]
LGTM
